src/Features/Lsif/Generator/Generator.cs

```diff
@@ -166,10 +166,11 @@ public static Generator CreateAndWriteCapabilitiesVertex(ILsifJsonWriter lsifJso
                 {
                     if (tasks[i].IsFaulted)
                     {
+                        var documentExceptionMessage = $"Exception while processing {documents[i].FilePath}";
                         var exception = tasks[i].Exception!.InnerExceptions.Single();
```

Comment:
>Single [](http://example.com/codeflow?start=76&length=6)

I assume it's not possible for there to be more than one exception in InnerExceptions?

Reply:
Task.Exception is always an AggregateException, I'm guessing because of things like Task.WhenAll() or something. But for a regular faulted task that came from Task.Run(), I don't know how there'd be a second exception -- once the first is thrown nothing is running to produce a second one!